AUMC_Example/pre_MEDS.py

```diff
@@ -0,0 +1,207 @@
+"""
+import rootutils
+
+root = rootutils.setup_root(__file__, dotenv=True, pythonpath=True, cwd=True)
+
+import gzip
+from collections.abc import Callable
+from datetime import datetime
+from pathlib import Path
+
+import hydra
+import polars as pl
+from loguru import logger
+from omegaconf import DictConfig, OmegaConf
+
+from MEDS_transforms.utils import get_shard_prefix, hydra_loguru_init, write_lazyframe
+
+ADMISSION_ID = "admissionid"
+PATIENT_ID = "patientid"
+
+def load_raw_aumc_file(fp: Path, **kwargs) -> pl.LazyFrame:
+    """Load a raw AUMCdb file into a Polars DataFrame.
+
+    Args:
+        fp: The path to the AUMCdb file.
+
+    Returns:
+        The Polars DataFrame containing the AUMCdb data.
+    """
+
+    return pl.scan_csv(fp, infer_schema_length=10000000, encoding="utf8-lossy", **kwargs)
+
+
+def process_patient_and_admissions(df: pl.LazyFrame) -> pl.LazyFrame:
+    """Takes the admissions table and converts it to a form that includes timestamps.
+
+    As AUMCdb stores only offset times, note here that we add a CONSTANT TIME ACROSS ALL PATIENTS for the true
+    timestamp of their health system admission. This is acceptable because in AUMCdb ONLY RELATIVE TIME
+    DIFFERENCES ARE MEANINGFUL, NOT ABSOLUTE TIMES.
+
+    The output of this process is ultimately converted to events via the `patient` key in the
+    `configs/event_configs.yaml` file.
+    """
+
+    origin_pseudotime = pl.datetime(
+        year = pl.col("admissionyeargroup").str.extract(r"(2003|2010)").cast(pl.Int32),
+        month = 1, day = 1
+    )
+
+    # TODO: consider using better logic to infer date of birth for patients 
+    #       with more than one admission.
+    age_in_years = ((
+        pl.col("agegroup").str.extract("(\\d{2}).?$").cast(pl.Int32) + 
+        pl.col("agegroup").str.extract("^(\\d{2})").cast(pl.Int32)
+    ) / 2).ceil()
+    age_in_days = age_in_years * 365.25
+    # We assume that the patient was born at the midpoint of the year as we don't know the actual birthdate
+    pseudo_date_of_birth = origin_pseudotime - pl.duration(days=(age_in_days - 365.25 / 2))
+    pseudo_date_of_death = origin_pseudotime + pl.duration(milliseconds=pl.col("dateofdeath"))
+
+
+    return df.filter(pl.col("admissioncount") == 1).select(
+        PATIENT_ID, 
+        pseudo_date_of_birth.alias("dateofbirth"),
+        "gender",
+        origin_pseudotime.alias("firstadmittedattime"),
+        pseudo_date_of_death.alias("dateofdeath")
+    ), df.select(PATIENT_ID, ADMISSION_ID)
+
+
+def join_and_get_pseudotime_fntr(
+    table_name: str,
+    offset_col: str | list[str],
+    pseudotime_col: str | list[str],
+    output_data_cols: list[str] | None = None,
+    warning_items: list[str] | None = None,
+) -> Callable[[pl.LazyFrame, pl.LazyFrame], pl.LazyFrame]:
+    """Returns a function that joins a dataframe to the `patient` table and adds pseudotimes.
+
+    Also raises specified warning strings via the logger for uncertain columns.
+
+    TODO
```

Comment:
You should fill this out

Reply:
@prockenschaub do you have feedback on this?